.gitlab-ci.yml

```
@@ -126,6 +126,32 @@ test-linux-stable:                 &test-linux
     - time cargo test --all --release --verbose --locked
     - sccache -s
 
+
+test-srml-staking:                 &test-srml-staking
+  stage:                           test
+  <<:                              *docker-env
+  variables:
+    # Enable debug assertions since we are running optimized builds for testing
+    # but still want to have debug assertions.
+    RUSTFLAGS: -Cdebug-assertions=y
+    RUST_BACKTRACE: 1
+  except:
+    variables:
+      - $DEPLOY_TAG
+  only:
+    changes:
+      - .gitlab-ci.yml
+      - srml/staking/*
+  script:
+    - cd srml/staking/
+    - time cargo test --tests tests --no-default-features --features std
```

```suggestion
    - time cargo test --no-default-features --features std
```
yes actually for now only relevant test of staking are in tests modules, but might be better to test for whole package in case in the future we add test in lib.rs that have different behavior depending on this feature.
(how is time important here, I mean in my current other PR I have inflation module which doesn't need to be tested with both with and without the feature, does it worth filter out my module in this test ?)

---

I don't understand entirely. With the current version of the pr the test is only run when there are changes detected in the `srml/staking` subdirectory. Do you think it should also be run for changes to other directories of the source tree?

---

Not really, this is completely isolated to the `srml/staking`. Though I don't see it as a big deal to run the tests twice all the time, and not only there is a change?

---

theoritically the behavior of the module could change if its dependencies changes and dependency can change without modfiying srml/staking/* files due to current versionning using path.

But changing the behavior of the version without the feature and not the version with the feature seems very very unlikely. Hmm so I would say it is fine this way.


Oh also I forget but that's probably better to run them with release I think

```suggestion
    - time cargo test --release --no-default-features --features std
```

---

I cannot commit this, but sounds totally reasonable

---

and do I understand correctly that this test is already run from the `cargo test` from the top-level directory but with a different "feature"?

---

yes correct, cargo test --all will run this crate tests with default features which include "equalize" feature.

this line run the test suite of this crate but without the "equalize" feature